Python/ast.c

```diff
@@ -619,7 +621,9 @@ validate_pattern(struct validator *state, pattern_ty p, int star_ok)
                     break;
                 }
             }
-
+            if (ret == 0) {
+                break;
+            }
```

Comment:
We need another such break in line 616 (after the while(1) loop). Will add.

Reply:
I could add the assert at the beginning of each of the validate_* functions.

Reply:
> I could add the assert at the beginning of each of the validate_* functions.

Not sure I follow. In my mental model, either the code path causing an error at that particular spot is exercised by the test suite, or it isn't. How does adding more asserts at different places help catch it? Is that third error code path already exercised, but exercised in a way that either doesn't end up calling `_PyAST_Compile`, or ends up silencing the error before it does?

Reply:
The code path is hit in the test, but then the for loop doesn't run because ``asdl_seq_LEN(p->v.MatchClass.kwd_attrs) == 0`` and so the ``if(ret == 0)`` I added after the for loop exits with the error.  I'll try to add a test that does go into the for loop, and then the new assert should trip.